src/gcodeExport.cpp

```diff
@@ -246,10 +249,6 @@ void GCodeExport::setFlavor(EGCodeFlavor flavor)
     {
         firmware_retract = true;
     }
-    else 
-    {
-        firmware_retract = false;
-    }
```

Comment:
Let me give you some concrete examples of how Marlin behaves.

Firmware retract can be enabled at compile time for a specific machine, but that does not imply volumetric.  Therefore without this patch the extrusion rates are wrong without this patch since currently the only way to enable firmware retract is to switch on volumetric.

Volumetric mode on Marlin is enabled by specifying the filament diameter with, e.g. "M200 D1.75" and turned off with "M200 0" .  I have another patch to Cura that looks at the machine_start_code for an M200... and then switches then switches the machine_gcode_flavor to volumetric which allows CuraEngine to do the right thing.

Bottom line is machine_firmware_retract and volumetric have to be treated independently.